src/constraint_aware_append.c

```diff
@@ -180,18 +195,23 @@ ca_append_begin(CustomScanState *node, EState *estate, int eflags)
 			case T_WorkTableScan:
 			case T_ForeignScan:
 			case T_CustomScan:
-
-				/*
-				 * If this is a base rel (chunk), check if it can be excluded
-				 * from the scan. Otherwise, fall through.
-				 */
-				if (rte->rtekind == RTE_RELATION &&
-					rte->relkind == RELKIND_RELATION &&
-					!rte->inh &&
-					excluded_by_constraint(rte, appinfo, restrictinfos))
-					break;
+				{
```

Comment:
This could also benefit from being moved to a function, something like:

```
if (is_excluded_chunk_scan(node, lfirst(lc_info)))
    break;
```

Reply:
Refactored this section, but not sure if it was worth it, because we have to pass in a lot of auxiliary variables.

Reply:
I see. But at least this switch is a bit more readable and the function name is helpful in understanding what the rather complicated boolean statement does. Makes it self-documenting.